Fix split/netrw window close handling

rbong opened the pull request:
There were window closing handling errors introduced with #22.

When opening netrw the folder buffer is deleted just before opening neovim. This prevents vifm from being reopened when exiting, but as an unintended side-effect it closes the current window. This has been fixed.

When exiting from a non-split window, the current buffer is deleted and the window is closed. Now, the window is closed only if vifm was opened in a split, the window is closed.

This fixes #25

rbong commented:
I'm not sure why changes that have already been merged are showing in the diff. I will clean up my history and that should fix it.

rbong commented:
Fixed the weird diff issues, should now be ready to merge if you're happy with it

rbong commented:
I realized that this and some other changes break switching back to the alternate buffer. I've changed it so that when closing Vifm it will return to the original buffer if appropriate.

xaizek commented on plugin/vifm.vim:
```diff
@@ -122,9 +119,6 @@ function! s:StartVifm(mods, count, editcmd, ...)
 				silent! bdelete! #
-				if self.split
-					close
-				endif
```
Without these three lines in neovim (but not in Vim) I get split window left on the screen, so one more window is created after each `:Vifm` with `:let g:vifm_embed_split = 1`.

rbong commented:
It's clear I need to really carefully test this.
I have a resolution I believe. It was a careful mix of all of my previous tweaks.

I've performed all of the following tests. If there was a mistake I fixed it and repeated all of the tests.

All of the following settings were tested:
  - Split off
  - Split on
  - netRW replacement

For all of the settings, all of the following things were tested:
  - In a single window with no buffer, vifm can be opened and closed without errors or exiting vim
  - In a single window with no buffer, vifm can open files without issue
  - In a single window with an alt file, vifm can be opened and closed without overwriting the alt file
  - In a single window with an alt file, vifm can open new files without issue
  - In multiple windows with no buffer, vifm can be opened and closed without closing the window
  - In multiple windows with no buffer, vifm can open files without issue
  - In multiple windows with an alt file, vifm can be opened and closed without overwriting the alt file or closing the window
  - In multiple windows with an alt file, vifm can open new files without issue

Additional tests:
  - Split on
    - The split does not persist
  - netRW replacemenet
    - Folder does not persist
    - Can open and exit without issue using "e ." (same tests as above)
    - Can open and exit without issue using "vs ." (same tests as above)

All tests were repeated in vim and neovim.

I have also cleaned up my branch history.

Unrelated issues encountered:
  - Cannot define two separate embedded instances of Vifm (Vim), I will fix this in a separate PR as it seems to not have been introduced by #22

xaizek commented on plugin/vifm.vim:
```diff
@@ -110,22 +110,33 @@ function! s:StartVifm(mods, count, editcmd, ...)
 				let data = b:data
 				buffer #
```
Looks like this line also needs to be updated to become
```vim
				if bufnr('%') == bufnr('#') && !data.split
					enew
				else
					buffer #
				endif
```
Otherwise alternative file isn't preserved on:
1. `:e something`
2. `:ene`
3. `:Vifm`
4. `:q` (in Vifm)

rbong replied:
I'll change and re-test, plus this gives me a few new tests to try

rbong replied:
This should now be resolved. I repeated all of the tests above and additionally tested preserving alt empty buffers with single-window and multi-window setups. I tested neovim to make sure it wasn't suffering the same issue in any cases.

xaizek commented:
Great. Thanks a lot, @rbong!
